src/modals/WalletConnect/WalletConnectSessionProposalModal.tsx

```diff
@@ -0,0 +1,200 @@
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { AlertTriangle, PlusSquare } from 'lucide-react'
+import { usePostHog } from 'posthog-js/react'
+import { useEffect, useState } from 'react'
+import { Trans, useTranslation } from 'react-i18next'
+import styled from 'styled-components'
+
+import AssetLogo from '@/components/AssetLogo'
+import InfoBox from '@/components/InfoBox'
+import AddressSelect from '@/components/Inputs/AddressSelect'
+import { Section } from '@/components/PageComponents/PageContainers'
+import Paragraph from '@/components/Paragraph'
+import { useAppDispatch, useAppSelector } from '@/hooks/redux'
+import useAddressGeneration from '@/hooks/useAddressGeneration'
+import CenteredModal, { ModalFooterButton, ModalFooterButtons } from '@/modals/CenteredModal'
+import { selectAddressesInGroup } from '@/storage/addresses/addressesSelectors'
+import { saveNewAddresses } from '@/storage/addresses/addressesStorageUtils'
+import { networkPresetSwitched } from '@/storage/settings/networkActions'
+import { Address } from '@/types/addresses'
+import { SessionProposalEvent } from '@/types/walletConnect'
+import { getRandomLabelColor } from '@/utils/colors'
+import { cleanUrl } from '@/utils/misc'
+import { isNetworkValid, parseSessionProposalEvent } from '@/utils/walletConnect'
+
+interface WalletConnectSessionProposalModalProps {
+  approveProposal: (signerAddress: Address) => Promise<void>
+  rejectProposal: () => Promise<void>
+  proposalEvent: SessionProposalEvent
+  onClose: () => void
+}
+
+const WalletConnectSessionProposalModal = ({
+  onClose,
+  approveProposal,
+  rejectProposal,
+  proposalEvent
+}: WalletConnectSessionProposalModalProps) => {
+  const { t } = useTranslation()
+  const posthog = usePostHog()
+  const currentNetworkId = useAppSelector((s) => s.network.settings.networkId)
+  const currentNetworkName = useAppSelector((s) => s.network.name)
+  const dispatch = useAppDispatch()
+  const { requiredChainInfo, metadata } = parseSessionProposalEvent(proposalEvent)
+  const addressesInGroup = useAppSelector((s) => selectAddressesInGroup(s, requiredChainInfo?.addressGroup))
+  const { generateAddress } = useAddressGeneration()
+
+  const [signerAddress, setSignerAddress] = useState<Address>()
+
+  const group = requiredChainInfo?.addressGroup
+
+  const showNetworkWarning =
+    requiredChainInfo?.networkId && !isNetworkValid(requiredChainInfo.networkId, currentNetworkId)
+
+  useEffect(() => {
+    setSignerAddress(
+      addressesInGroup.length > 0 ? addressesInGroup.find((a) => a.isDefault) ?? addressesInGroup[0] : undefined
+    )
+  }, [addressesInGroup])
+
+  const handleSwitchNetworkPress = async () => {
```

Comment:
This method is async but nothing inside is awaited

Reply:
Same here 😇